packages/nightingale-track/src/nightingale-track.ts

```diff
@@ -203,16 +183,18 @@ class NightingaleTrack extends withManager(
     }
   }
 
-  #getPTMResidueShape(f: PTM & { feature: Feature; position: number }) {
-    let ptmLength = 1;
-    // For longer proteins, the PTMs have to be shown prominent in the first look
-    if (this.length && this.length > 500) {
-      ptmLength = this.getSingleBaseWidth() < 4 ? 4 : 1;
-    }
+   #getResidueShape(f: Residue & { feature: Feature; position: number }) {
+    let residueLength = 1;
+     // Below logic is to show it prominent for longer proteins until the point where residueLength is enough to be visible on itself.
+     const optimalWidth = 6;
+     const widthDifference = optimalWidth - this.getSingleBaseWidth();
+     if (this.getSingleBaseWidth() < optimalWidth && widthDifference > residueLength) {
```

Comment:
Could just assign to a variable eg `const singleBaseWidth = this.getSingleBaseWidth()` rather than calling twice.